wrappers/binaries.go

```diff
@@ -21,13 +21,57 @@
 package wrappers
 
 import (
+	"io"
 	"os"
+	"regexp"
+	"strconv"
+	"strings"
 
 	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/snap"
 )
 
+func detectLegacy() (noCompletion bool, legacyCompletion bool) {
+	fd, err := os.Open(dirs.BashCompletionScript)
+	if err != nil {
+		return true, false
+	}
+	defer fd.Close()
+	buf := new(strings.Builder)
+	io.Copy(buf, fd)
+	bash_completion := buf.String()
```

Comment:
hmm, so we load all of the file to a buffer which then returns it at a string, this could be done by:
```
d, err := ioutil.ReadFile(dirs.BasCompletionScript)
// check err
bash_completion := string(d)
```
but then we look a match inside the file, so perhaps we don't really need to load all 76kB of that file, but rather read it line by line looking for a match?

```
reRelease := regexp.MustCompile(`(?:^|\n)# *RELEASE: ([0-9]+)[.]([0-9]+)(?:$|\n)`)
reBashCompletionVersinfo := regexp.MustCompile(`(?:^|\n)BASH_COMPLETION_VERSINFO=`)
s := bufio.NewScanner(fd)
for s.Scan() {
    line := s.Text()
    // process
}
if err := s.Err(); err != nil {
    return fmt.Errorf("cannot read file: %v", err)
}
```

Reply:
Changed use the line scanner.